### ISSUE TITLE
Add another level to API docs contents?

### ISSUE THREAD

**Issue by ivirshup** [shapely/shapely#1503]
I have a feature request for the documentation.

Could another level be added to the table of contents for the API docs? I am having trouble navigating the new docs when I'm not able to easily see a listing of classes or jump to any particular one.

As an example, I would like to see what the various Geometry types are at a glance, but only see this:

![image](https://user-images.githubusercontent.com/8238804/186664025-33e6ea66-653b-46b2-a017-2d248846892e.png)

**Comment by jorisvandenbossche** [comment]
@ivirshup Thanks for opening the issue!

I agree that it would be nice to have an easier overview of those in the left or right sidebar (similarly on the other pages, eg to get a quick overview of all predicates functions on that page: https://shapely.readthedocs.io/en/latest/predicates.html)

However, this is a bit of a limitation of how we use sphinx / autodoc in this case. The content that is being generated by autodoc doesn't create any titles or sections or similar, and the content in the sidebar of the sphinx theme is based on those (it are just some `<dl>` (definition list) html objects, although they have an `id` to link to, they don't make the structure of the document).

**Comment by jorisvandenbossche** [comment]
One alternative I have been thinking about is that we could also create a single page per class or per function. And then we could use `autosummary` to create an easier overview of all functions. 

Similarly to how eg numpy does it here: https://numpy.org/doc/stable/reference/arrays.indexing.html (and then a single function getting a page of its own, for example https://numpy.org/doc/stable/reference/generated/numpy.take.html).

This is actually also similar to the PostGIS function documentation (for example https://postgis.net/docs/ST_Contains.html).

@caspervdw since you added those reference docs initially, what would you think about such an approach?

**Comment by ivirshup** [comment]
If you wanted to get around `autodoc` here, I think you could also create a template which creates headers for each item you'd like to have in the table of contents. Here's [a template](https://github.com/scverse/scvi-tools/blob/master/docs/_templates/autosummary/class.rst) which does this, and [the resulting docs](https://docs.scvi-tools.org/en/stable/api/reference/scvi.model.PEAKVI.html#scvi.model.PEAKVI).

**Comment by jorisvandenbossche** [comment]
Yes, but so as far as I understand, that's a template for autosummary when creating a page per function, and this template is not used by autodoc when bundling them all on a single page. 

So what you mention is exactly what I wanted to suggest above as well to use autosummary with separate pages per function (in numpy/pandas, we are using a similar template).

**Comment by jorisvandenbossche** [comment]
I opened a PR experimenting with autosummary -> https://github.com/shapely/shapely/pull/1529

**Comment by jbbqqf** [comment]
Hi @ivirshup — drive-by triage of stale issues: I believe this one can be closed as resolved by [#1529](https://github.com/shapely/shapely/pull/1529).

@jorisvandenbossche opened PR #1529 ("DOC: autogenerate separate reference pages per function with autosummary") in this very thread and it was merged 2022-09-24. The pattern is now applied across the API reference docs — see for example [`docs/geometry.rst`](https://github.com/shapely/shapely/blob/main/docs/geometry.rst) where each Geometry subclass gets its own page via:

```rst
.. autosummary::
    :toctree: reference/

    Point
    LineString
    LinearRing
    Polygon
    MultiPoint
    MultiLineString
    MultiPolygon
    GeometryCollection
```

The same `:toctree: reference/` pattern is used in `docs/predicates.rst`, `docs/measurement.rst`, `docs/set_operations.rst`, etc., so the deeper TOC navigation you asked for (an individual page per class/function reachable from the sidebar) is now in place — e.g. [shapely.Polygon](https://shapely.readthedocs.io/en/stable/reference/shapely.Polygon.html).

No activity on this issue since 2022-09. Probably safe to close.

---
_Drive-by triage from an outside contributor (data engineer GCP/Python, [jbbqqf](https://github.com/jbbqqf)). Comment drafted with Claude assistance; evidence and conclusion are mine._
